packages/contracts-bedrock/scripts/L2Genesis.s.sol

```diff
@@ -392,8 +397,91 @@ contract L2Genesis is Script, Artifacts {
     }
 
     //////////////////////////////////////////////////////
-    /// Helpers
+    /// Other
     //////////////////////////////////////////////////////
+
+    function _getHardforkTime(uint256 _hardforkTimeOffset, uint256 _genesisTime) internal pure returns (uint256) {
+        uint256 _default = 0;
+        if (_hardforkTimeOffset > 0) {
+            return _genesisTime + _hardforkTimeOffset;
+        }
+        return _default;
+    }
+
+    function _getChainConfig() internal returns (string memory) {
+        // TODO Go code is using block.Time() from provided *types.Block.
+        // devnetL1-template.json specifies "l2BlockTime": 2,
+        // so should genesisTime = cfg.l2BlockTime?
+        // block.timestamp = 1
+        uint256 genesisTime = block.timestamp;
```

Comment:
Using `block.timestamp` is a good way to get the time, it means we will need an RPC url pointing to L1 when generating the genesis json which is similar to how we do it today, that is how we get the `types.Block`

Reply:
I think [this](https://github.com/ethereum-optimism/optimism/pull/9328/files#diff-2b05d0be8d7d8de7870ac51225a0728adeb9a6df9819eaa654c455a23b8f509eR416-R433) is correct, at least compared to [this Go code](https://github.com/ethereum-optimism/optimism/blob/9cca805d5e46746c60070d12bec94e1d93c4dcf3/op-chain-ops/genesis/genesis.go#L64-L69).

The [Go code checks](https://github.com/ethereum-optimism/optimism/blob/9cca805d5e46746c60070d12bec94e1d93c4dcf3/op-chain-ops/genesis/config.go#L464-L473) if each specific time offset is `nil`, and returns `nil` if so  - I assume this is to disable the hardfork and not have it included in the genesis output. I tried to replicate this behavior by [defaulting the time offset](https://github.com/ethereum-optimism/optimism/pull/9328/files#diff-fcedbe6005fe092aab124a65c9eb9a2c5824c1cd83461b144daea68b6c8a4d2bR171-R174) to `-1` if it's not found in the deploy config and by checking if it's `-1` in `L2Genesis.s.sol` so it's not included it in the genesis output